Improve docs on create_mode for sql_database (#3599)
terraform-providers/terraform-provider-azurerm: website/docs/r/sql_database.html.markdown

* `server_name` - (Required) The name of the SQL Server on which to create the database.

> * `create_mode` - (Optional) Specifies how to create the database. Defaults to `Default`, ie. create a new database.  This can also be set to `PointInTimeRestore` to restore from a snapshot.

Reviewer: ```suggestion
* `create_mode` - (Optional) Specifies how to create the database. Must be either `Default` to create a new database or `PointInTimeRestore` to restore from a snapshot. Defaults to `Default`.
```
